.github/workflows/ci-python-sdk.yaml

```diff
@@ -12,6 +12,7 @@ on:
       - 'python-sdk/**'
```

Comment:
@pankajastro please, share your findings about why this PR has
`28 successful and 6 skipped checks`

When compared to #924
`24 successful and 1 skipped checks` 

Why did the checks increase so much?

Reply:
Looking..., yes ideally it should have 25 successful and 1 skipped

Reply:
Look like some job from the event `pull_request_target` is also getting executed and if the condition is not matching then getting skipped but still it strange since we have added branch filter

Reply:
Also, I can see some jobs are double for example 

`Build and test astro / Type-Check (pull_request_target) ` and 
`Build and test astro / Type-Check (pull_request_target`

Reply:
@tatiana this look like a pre-existing issue in our CI we add any label in our PR it ran couple job twice only by `pull_request` and other by `pull_request_target`

Reply:
we fixed some issues of running twice here https://github.com/astronomer/astro-sdk/pull/950 but I feel we still need to re-investigate how we should run CI on the fork